src/helpers/model/filters.ts

```diff
@@ -1,49 +1,140 @@
 // (C) 2018 GoodData Corporation
-import { VisualizationObject } from '@gooddata/typings';
+import { VisualizationInput } from '@gooddata/typings';
 import { getQualifierObject } from './utils';
 
-export const positiveAttributeFilter = (
+export function positiveAttributeFilter(
     qualifier: string,
-    inValues: string[]
-): VisualizationObject.IVisualizationObjectPositiveAttributeFilter => ({
-    positiveAttributeFilter: {
-        displayForm: getQualifierObject(qualifier),
-        in: inValues
-    }
-});
+    inValues: string[],
+    textFilter?: boolean
+): VisualizationInput.IPositiveAttributeFilter {
+    return {
+        positiveAttributeFilter: {
+            displayForm: getQualifierObject(qualifier),
+            in: inValues,
+            textFilter
+        }
+    };
+}
 
-export const negativeAttributeFilter = (
+export function negativeAttributeFilter(
     qualifier: string,
-    notInValues: string[]
-): VisualizationObject.IVisualizationObjectNegativeAttributeFilter => ({
-    negativeAttributeFilter: {
-        displayForm: getQualifierObject(qualifier),
-        notIn: notInValues
-    }
-});
+    notInValues: string[],
+    textFilter?: boolean
+): VisualizationInput.INegativeAttributeFilter {
+    return {
+        negativeAttributeFilter: {
+            displayForm: getQualifierObject(qualifier),
+            notIn: notInValues,
+            textFilter
+        }
+    };
+}
 
-export const absoluteDateFilter = (
+export function absoluteDateFilter(
     dataSet: string,
     from?: string,
     to?: string
-): VisualizationObject.IVisualizationObjectAbsoluteDateFilter => ({
-    absoluteDateFilter: {
-        dataSet: getQualifierObject(dataSet),
-        from,
-        to
-    }
-});
+): VisualizationInput.IAbsoluteDateFilter {
+    return {
+        absoluteDateFilter: {
+            dataSet: getQualifierObject(dataSet),
+            from,
+            to
+        }
+    };
+}
 
-export const relativeDateFilter = (
+export function relativeDateFilter(
     dataSet: string,
     granularity: string,
     from?: number,
     to?: number
-): VisualizationObject.IVisualizationObjectRelativeDateFilter => ({
-    relativeDateFilter: {
-        dataSet: getQualifierObject(dataSet),
-        granularity,
-        from,
-        to
+): VisualizationInput.IRelativeDateFilter {
+    return {
+        relativeDateFilter: {
+            dataSet: getQualifierObject(dataSet),
+            granularity,
+            from,
+            to
+        }
+    };
+}
+
+export class AttributeFilterBuilder {
+    constructor(private readonly qualifier: string) {
+        this.qualifier = qualifier;
```

Comment:
This is not needed since you specified `private readonly qualifier: string` in constructor parameters.

```ts
constructor(private readonly x: string) { }
// is the same as
private readonly x: string;
constructor(x: string) {
    this.x = x;
}
```
However, not many people are familiar with the first version so I'd probably use the second (more verbose) one, just to be sure everyone knows what is going on... [Blog post](https://www.stevefenton.co.uk/2013/04/Stop-Manually-Assigning-TypeScript-Constructor-Parameters/)

Reply:
yep that was my motivation as well. empty constructor looks awkward.